### PR TITLE
feat(tests): added execute deploy in any status

### PR DESCRIPTION
**Pull request by nestoragent**
# Contribution checklist

## Brief description

feat(tests): added execute deploy in any status

## Checklist

* [ ] - UI test added to added/changed functionality?
* [ ] - Screenshot is added to PR if applicable ?
* [ ] - Does a local formatting is applied (rush format)
* [ ] - Does a local svelte-check is applied (rush svelte-check)
* [ ] - Does a local UI tests are executed [UI Testing](../tests/readme.md)
* [ ] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [ ] - Does the translations are up to date?
* [ ] - Does it well tested?
* [ ] - Tested for Chrome.
* [ ] - Tested for Safari.
* [ ] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [ ] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [ ] - Are required links are linked to PR?
* [ ] - Does new code is well documented ?

## Related issues

A list of closed updated issues

## Contributor requirements

* [ ] - Sign-off is provided. [DCO](https://github.com/apps/dco)
* [ ] - GPG Signature is provided. [GPG](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -221,13 +221,13 @@ jobs:
...
         if: always()
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
           name: playwright-results
           path: ./tests/sanity/playwright-report/
       - name: Get Allure history
         uses: actions/checkout@v4
-        if: ${{ github.ref == 'refs/heads/main' }}
+        if: ${{ github.ref == 'refs/heads/main' }} && always()
```
Comment (haiodo) [inline .github/workflows/main.yml]: Nope, it should be
```suggestion
        if: ${{ github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v') }}
```

Comment (nestoragent) [reply .github/workflows/main.yml]: great point, thank you!
Also need to add status checks, doing it =)

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -221,13 +221,13 @@ jobs:
...
         if: always()
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
           name: playwright-results
           path: ./tests/sanity/playwright-report/
       - name: Get Allure history
         uses: actions/checkout@v4
-        if: ${{ github.ref == 'refs/heads/main' }}
+        if: ${{ github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v') }} && always()
```
Comment (haiodo) [inline .github/workflows/main.yml]:
```suggestion
        if: ${{ github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v') }}
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -243,20 +243,20 @@ jobs:
...
       - name: Upload allure test results
         if: always()
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
           name: allure-report
           path: ./allure-report/
       - name: Deploy report to Github Pages
-        if: ${{ github.ref == 'refs/heads/main' }}
+        if: ${{ github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v') }} && always()
```
Comment (haiodo) [inline .github/workflows/main.yml]:
```suggestion
        if: ${{ github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v') }}
```

**Review by haiodo — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 3 above)
Comment (nestoragent) [reply .github/workflows/main.yml]: fixed =)
thank you!

File: .github/workflows/main.yml (hunk 2 above)
Comment (nestoragent) [reply .github/workflows/main.yml]: fixed =)
thank you!

**Review by haiodo — APPROVED** [review]

**Pull request merged**